src/components/DateTimePicker/__test__/dateTimePicker.spec.js

```diff
@@ -0,0 +1,110 @@
+                value={value}
+                onFocus={onFocusMockFn}
+            />,
+        );
+        component.find('input').simulate('focus');
+        component.find('input').simulate('blur');
+        expect(onFocusMockFn).toHaveBeenCalledWith(value);
+    });
+    it('should set isOpen to true and fire onClick when readOnly is not passed', () => {
+        const onClickMockFn = jest.fn();
+        const component = mount(
+            <DateTimePicker
+                label="unit-testing-dateTimePicker"
+                value={value}
+                onClick={onClickMockFn}
+            />,
+        );
+        component.find('input').simulate('click');
+        expect(onClickMockFn).toHaveBeenCalledTimes(1);
+        expect(component.find('DateTimePickerModal').prop('isOpen')).toBe(true);
+    });
+    it('should not set isOpen to true and not fire onClick when readOnly is passed', () => {
+        const onClickMockFn = jest.fn();
+        const component = mount(
+            <DateTimePicker
+                readOnly
+                label="unit-testing-dateTimePicker"
+                value={value}
+                onClick={onClickMockFn}
+            />,
+        );
+        component.find('input').simulate('click');
+        expect(onClickMockFn).toHaveBeenCalledTimes(0);
+        expect(component.find('DateTimePickerModal').prop('isOpen')).toBe(false);
+    });
+    it('should open the modal when enter key or space key is pressed while input is focused and readOnly is not passed', () => {
+        const values = [ENTER_KEY, SPACE_KEY];
+        values.forEach(keyCode => {
+            const component = mount(<DateTimePicker label="unit-testing-dateTimePicker" />);
+            const input = component.find('input');
+            input.simulate('focus');
+            input.simulate('keyDown', { keyCode });
+            expect(component.find('DateTimePickerModal').prop('isOpen')).toBe(true);
+        });
+    });
+    it('should not open the modal when enter key or space key is pressed while input is focused and readOnly is passed', () => {
+        const values = [ENTER_KEY, SPACE_KEY];
+        values.forEach(keyCode => {
+            const component = mount(
+                <DateTimePicker label="unit-testing-dateTimePicker" readOnly />,
+            );
+            const input = component.find('input');
+            input.simulate('focus');
+            input.simulate('keyDown', { keyCode });
+            expect(component.find('DateTimePickerModal').prop('isOpen')).toBe(false);
+        });
+    });
+    it('should set the right input value when value change dynamically', () => {
+        const component = mount(
+            <DateTimePicker label="unit-testing-dateTimePicker" value={value} />,
+        );
+        component.setProps({ value: new Date(2019, 9, 24, 10, 48, 45) });
+        component.update();
+        expect(component.find('input').prop('value')).toBe('10/24/2019, 10:48 AM');
+    });
```

Comment:
let's change this test:
1. mock the formatDateTime helper
2. test that what is passed to input value is what is returned by the formatDateTime mock
3. test name: ```should set the right input value```